### ISSUE TITLE
Is :first Supported by Weasyprint or Not?

### ISSUE THREAD

**Issue by jwalkerinterpres** [Kozea/WeasyPrint#2088]
I am using Weasyprint version 60.2.

As [per this page](https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#css-paged-media-module-level-3) (emphasis added), the `:first` CSS selector is supported:

> All the features of this draft are available, including:
> 
> the @page rule and the :left, :right, **<ins>:first</ins>** and :blank selectors;

However, when I try to use the `:first` selector, eg.

    td div:first {

I get an error from Weasyprint:

    ('Unknown pseudo-class', 'first')

**Comment by grewn0uille** [comment]
Hello!

The `:first` pseudo-class matches the first printed page of a document. Which means it’s to be used with a `@page` rule like `@page :first` (select the first page).

If you want to select the first div of a td, you can use the `:first-of-type` pseudo-class like that `td div:first-of-type`.

**Comment by jwalkerinterpres** [comment]
Thanks, but the documentation could really use a clearer way for users to understand "will Weasyprint support selector X or not?"

**Comment by grewn0uille** [comment]
I’m not sure we can make clearer than

> All the features of this draft are available, including:
> the @ page rule and the :left, :right, :first and :blank selectors;

to indicate that the `:first` selector is supported. But if you have a better suggestion, don’t hesitate to open a PR :)

As the `:first` selector is to be used with `@page` rules, it’s normal that it’s not working in other contexts.

> Unfortunately [caniuse.com](http://caniuse.com/) doesn't have a Weasyprint entry :(

Yes unfortunately

**Comment by jwalkerinterpres** [comment]
>But if you have a better suggestion, don’t hesitate to open a PR :)

In a perfect world I could:

* have a question (eg. "Can I use `:first`?")
* go to the Weasyprint website 
* get that question answered, easily

Right now that last step isn't the case.  Imagine if there was ...

* a search box where I can type in a selector and see whether it's supported
* a page with a big list (or table) of every selector, and "supported" checkboxes next to the supported ones
* any other UI that lets me easily answer "Can Weasyprint do ____?"

However, I'm not prepared to commit to submitting a PR for such a page, so I'll stop at saying "this would be useful to have".

**Comment by grewn0uille** [comment]
Your problem is not that the selector is not supported, it’s that you’re not using it correctly. How would a large table help you in this case?

**Comment by jwalkerinterpres** [comment]
My apologies for the confusion.  I was actually looking for `:first-child`, but too many years of using jQuery had me thinking the selector was called just `:first`.  Dumb mistake.

However, the situation wasn't helped by `:first-child` not even being mentioned on https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#css-paged-media-module-level-3.  If Weasyprint's website had any sort of page like I described (search for selector, big list of selectors, etc.) I could go to that page, look for `:first`, and easily learn that the library supports `:first`, but not `:first-child`.

Instead, we got this ticket.  It could be solved for future users if a maintainer could just grab a big list of selectors from somewhere like the MDN, somehow mark which ones are/aren't supported, and then somehow publish said list.
